pkg/hubble/relay/observer/server.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package observer
+
+import (
+	"context"
+	"fmt"
+
+	observerpb "github.com/cilium/cilium/api/v1/observer"
+	relaypb "github.com/cilium/cilium/api/v1/relay"
+	"github.com/cilium/cilium/pkg/hubble/relay/pool"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+)
+
+// PeerLister is the interface that wraps the List method.
+type PeerLister interface {
+	// List returns a list of peers with active connections. If a peer cannot
+	// be connected to; its Conn attribute must be nil.
+	List() []pool.Peer
+}
+
+// PeerReporter is the interface that wraps the ReportOffline method.
+type PeerReporter interface {
+	// ReportOffline allows the caller to report a peer as being offline. The
+	// peer is identified by its name.
+	ReportOffline(name string)
```

Comment:
unrelated to the PR, but long-term `name string` may not scale. Accepting some sort of a selector struct, which includes `name string`, and perhaps other things in future